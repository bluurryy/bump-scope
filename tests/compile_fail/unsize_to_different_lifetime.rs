use bump_scope::{unsize_bump_box, BumpBox};

fn evil_unsize(boxed: BumpBox<[i32; 3]>) -> BumpBox<'static, [i32]> {
    unsize_bump_box!(boxed)
}

fn main() {}
