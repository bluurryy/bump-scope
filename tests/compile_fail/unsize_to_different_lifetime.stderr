error: lifetime may not live long enough
 --> tests/compile_fail/unsize_to_different_lifetime.rs:4:5
  |
3 | fn evil_unsize(boxed: BumpBox<[i32; 3]>) -> BumpBox<'static, [i32]> {
  |                ----- has type `BumpBox<'1, [i32; 3]>`
4 |     unsize_bump_box!(boxed)
  |     ^^^^^^^^^^^^^^^^^^^^^^^ returning this value requires that `'1` must outlive `'static`
  |
  = note: this error originates in the macro `unsize_bump_box` (in Nightly builds, run with -Z macro-backtrace for more info)
